docs/Explanation/ADR/adr-004-did.md

```diff
@@ -0,0 +1,208 @@
+
+- `context` - for json-ld specification
+- `id` - indicating the DID of the document
+- `controller` - a list of DID controllers for the DID document
+- `verificationMethods` - a list of verification method objects
+- `services` - a list of services
+
+We will support the following verification relationships:
+
+- `authentication`
+- `assertionMethod`
+- `keyExchange`
+- `invocationMethod`
+- `delegationMethod`
+
+
+##### [DID Operations](https://www.w3.org/TR/did-core/#method-operations)
+
+
+###### Create
+
+The publication on chain of a DID document is allowed only when the submitted DID document contains at least one verification method that is listed in the verification relationship `authentication` and whose `blockchainAccountID` matches the transaction signer for the address part (ignoring the chain prefix).
+
+###### Resolve/Verify
+
+The integrity of the DID documents stored on the ledger are guaranteed by the underlying blockchain protocol. 
+
+> TODO: anything to add about resolution? see [169#issuecomment-891189527](https://github.com/allinbits/cosmos-cash/issues/169#issuecomment-891189527)
+
+###### Update
+
+Amends to a DID document are allowed only to:
+
+- controllers of one of the verification method listed in the `authentication` verification relationship
+- controllers of one of the verification method listed in the `authentication` verification relationship of the DID document identified in the the controller of the current DID document 
+
+```
+example: 
+
+giving did did:cash:abc:1111 described by document
+
+{
+    "@context": ...
+    "id": "did:cash:abc:1111",           <--- DID 111 subject
+    "controller": [
+        "did:cash:abc:2222"              <--- Controller for the document describing did:cash:abc:1111 
+    ]
+    "verificationMethods" : [
+        "id": "did:cash:abc:1111#key-1",
+        "type": "EcdsaSecp256k1VerificationKey2019",
+        "controller": "did:cash:abc:1111",   <--- Controller for the verification method identified by did:cash:abc:1111#key-1
+        "blockchainAccountID": "cosmos1g4fdrtxjjvzxvazkmj7zgz634a4hn93nry0p9u"
+    ],
+    "authentication": [
+        "id": "did:cash:abc:1111#key-1"      <--- Authentication relationships for key did:cash:abc:1111#key-1
+    ]
+}
+
+and did:cash:abc:2222
+
+{
+    "@context": ...
+    "id": "did:cash:abc:2222",         <--- DID subject 
+    "verificationMethods" : [
+        "id": "did:cash:abc:2222#key-1",
+        "type": "EcdsaSecp256k1VerificationKey2019",
+        "controller": "did:cash:abc:2222", <--- Controller for the verification method identified by did:cash:abc:1111#key-1
+        "blockchainAccountID": "cosmos1ts9ejqg7k4ht2sm53hycty875362yqxqmt9grj",
+    ],
+    "authentication": [
+        "id": "did:cash:abc:2222#key-1"    <--- Authentication relationships for key did:cash:abc:2222#key-1
+    ] 
+}
+
+according to the rules the DID did:cash:abc:2222 is authorized to amend the DID document for did:cash:abc:1111  
```

Comment:
**praise:** The example is good, but I would move the explanatory text from below the example to introducing the example.